File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.swift
func getPalindrome(from name: String) -> String {
    var characterDictionary = [Character : Int]()
    
    Array(name).forEach { characterDictionary[$0, default: 0] += 1 }

    var halfPalindrome = "",
        centerCharacter = "",
        oddCase = 0
    
    for (key, value) in characterDictionary.sorted(by:<) {
        if !value.isMultiple(of:2) { 
            oddCase += 1
            if oddCase > 1 { return "I'm Sorry Hansoo" }
            centerCharacter = String(key)
        }
        halfPalindrome.append(String(repeating: key, count: value / 2))
    }
    
    
    return halfPalindrome + centerCharacter + String(halfPalindrome.reversed())
}

print(getPalindrome(from: readLine()!))

File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.py
from collections import defaultdict

alphabet_dictionary = defaultdict(int)
for alphabet in input():
    alphabet_dictionary[alphabet] += 1

def return_palindrome(dictionary) -> str:
    failure_result = "I'm Sorry Hansoo"
    half_palindrome = ""
    centerAlphabet = ""
    odd_case = 0
    
    for key, value in sorted(dictionary.items()):
        if not value % 2 == 0:
            odd_case += 1
            if odd_case > 1:
                return failure_result
            else:
                centerAlphabet = key
            
        half_palindrome += key * int(value / 2)
    
    return half_palindrome + centerAlphabet + half_palindrome[::-1]


print(return_palindrome(alphabet_dictionary))